asmcomp/amd64/emit.mlp

```diff
@@ -865,14 +865,21 @@ let emit_instr fallthrough i =
                          ConstLabel lbl))
       done;
       D.text ()
-  | Lsetuptrap lbl ->
-      I.call (label lbl)
-  | Lpushtrap ->
-      cfi_adjust_cfa_offset 8;
-      I.push r14;
-      cfi_adjust_cfa_offset 8;
-      I.mov rsp r14;
-      stack_offset := !stack_offset + 16
+  | Lpushtrap {lbl_handler} ->
+      let load_label_addr s arg =
+       (* CR mshinwell: this needs more testing *)
```

Comment:
I tested it when the compiler was configured with PIC ./configure --with-pic on x86_64-pc-linux-gnu target. It passes all tests in the testsuite. I also manually inspected the assembly&objdump it generated for a simple test, and it looks fine. 
With pic, we get instruction of the form "lea  0x22(%rip),%r14" and offset 0x22 is the handler.